cli/src/commands/task/take_more_labels.rs

```diff
@@ -0,0 +1,258 @@
+            })?;
+            statistics.add_target_comments(target_comment_batch.len());
+
+            let mut target_labellings_to_update =
+                get_labellings_with_assigned_label(target_comment_batch, &only_for_target_label);
+
+            if target_labellings_to_update.is_empty() {
+                continue;
+            }
+
+            // Fetch source labellings for our given ids
+            let source_annotated_comments = client
+                .get_labellings(
+                    &source_dataset.full_name(),
+                    target_labellings_to_update.keys(),
+                )
+                .chain_err(|| {
+                    ErrorKind::Client("Getting batch of source labellings failed.".into())
+                })?;
+
+            for source_annotated_comment in source_annotated_comments.into_iter() {
+                let AnnotatedComment {
+                    comment: Comment { uid, .. },
+                    labelling: source_labelling,
+                    ..
+                } = source_annotated_comment;
+                let source_assigned = source_labelling.unwrap_or_default().assigned;
+                if source_assigned.is_empty() {
+                    continue;
+                }
+                let Labelling {
+                    assigned: target_assigned,
+                    dismissed,
+                    ..
+                } = target_labellings_to_update
+                    .remove(&uid)
+                    .expect("Tried to make a new labelling for a comment we don't need to update.");
+
+                // Merge assigned labels from the two datasets
+                let assigned = merge_assigned_labels(
+                    target_assigned,
+                    source_assigned,
+                    &transferred_label_parent,
+                );
+                let new_labelling = NewLabelling {
+                    assigned,
+                    dismissed,
+                };
+
+                client
+                    .update_labelling(
+                        &target_dataset.full_name(),
+                        &uid,
+                        Some(&new_labelling),
+                        None,
+                    )
+                    .chain_err(|| ErrorKind::Client("Updating labelling failed.".into()))?;
+                statistics.add_labellings_updated(1);
+            }
+
+            let _progress = make_progress()?;
+        }
+    }
+    Ok(())
+}
+
+/// Collect target comment ids and labellings, which contain the given assigend label.
+fn get_labellings_with_assigned_label(
+    annotated_comments: Vec<AnnotatedComment>,
+    assigned_label_name: &LabelName,
+) -> HashMap<Uid, Labelling> {
+    let mut target_labellings_to_update = HashMap::new();
+    // Consume the comments here as we break them up into parts for reuse
+    for annotated_comment in annotated_comments.into_iter() {
+        let AnnotatedComment {
+            comment, labelling, ..
```

Comment:
Interesting `..` notation, not sure what it does? I guess it unpacks.

Reply:
Yes, it's saying 'there are other fields in this `AnnotatedComment` and I don't care about them'. Otherwise you would get an error saying there are fields that we have not assigned to a variable.